pkg/l2announcer/l2announcer_test.go

```diff
@@ -474,119 +467,17 @@ func TestPolicyRedundancy(t *testing.T) {
 	}, fix.announcer.selectedServices[svcKey].byPolicies)
 
 	// Assert Proxy Neighbor Entry still exists
-	r = fix.proxyNeighborTable.Reader(fix.stateDB.ReadTxn())
-	iter, err = r.Get(statedb.All)
-	assert.NoError(t, err)
-	entries = statedb.Collect[*tables.L2AnnounceEntry](iter)
+	rtx = fix.stateDB.ReadTxn()
+	iter, _ = fix.proxyNeighborTable.All(rtx)
+	entries = statedb2.Collect[*tables.L2AnnounceEntry](iter)
 	assert.Len(t, entries, 1)
 	assert.Equal(t, entries[0], &tables.L2AnnounceEntry{
-		IP:               net.ParseIP(svc.Spec.ExternalIPs[0]),
-		NetworkInterface: policy.Spec.Interfaces[0],
-		Origins:          []resource.Key{svcKey},
-		Deleted:          false,
-		Revision:         1,
-	})
-
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	fix.announcer.jobgroup.Stop(ctx)
-	cancel()
-}
-
-// Test that when the last policy selecting a service is deleted, the selected service and proxy neighbor entries go
-// away.
-func TestPolicySoftDeleteService(t *testing.T) {
```

Comment:
Do we need to revisit this test after this PR?

Reply:
We shouldn't have to. The old statedb had no way of tracking deletes so we had a custom soft delete mechanism which is just gone now, so we don't specifically have to test it anymore.